docs/checklist_templates/eip_testing_checklist_template.md

```diff
@@ -0,0 +1,289 @@
+# EIP Execution Layer Testing Checklist Template
+
+Depending on the changes introduced by an EIP, the following template is the minimum baseline to guarantee test coverage of the Execution Layer features.
+
+## General
+
+- [ ] Code coverage
+    - [ ] Run produced tests against [EELS](https://github.com/ethereum/execution-specs) and verify that line code coverage of new added lines for the EIP is 100%, with only exeptions being unreachable code lines.
+    - [ ] Optional - Run against a second client and verify sufficient code coverage over new code added for the EIP.
+- [ ] Fuzzing
+    - [ ] TBD
+
+## New Opcode
+
+The EIP introduces one or more new opcodes to the EVM.
+
+### Test Vectors
+
+- [ ] Memory expansion
+    - [ ] Verify that the opcode execution results in the correct memory expansion, being by offset or size or interaction of both parameters (Size of zero should never result in memory expansion, regardless of offset value). Test at least the following memory expansion sizes
+        - [ ] Zero bytes expansion
+        - [ ] Single byte expansion
+        - [ ] 31 bytes expansion
+        - [ ] 32 bytes expansion
+        - [ ] 33 bytes expansion
+        - [ ] 64 or more bytes expansion
+- [ ] Stack over/underflows
+    - [ ] If the opcode pushes one or more items to the stack, and the opcode pushes more elements than it pops, verify that the opcode execution results in exeptional abort when pushing elements to the stack would result in the stack having more than 1024 elements.
+    - [ ] If the opcode pops one or more items to the stack, or it has a minimum stack height of one or more, verify that the opcode execution results in exeptional abort then stack has 1 less item than the minimum stack height expected.
+- [ ] Execution context
+    - [ ] Normal call
+    - [ ] Static call
+        - [ ] Verify exeptional abort if the opcode attempts to modify the code, storage or balance of an account
+    - [ ] Delegate call
+        - [ ] If the opcode modifies the storage of the account currently executing it, verify that only the account that is delegating execution is the one that has its storage modified.
+        - [ ] If the opcode modifies the balance of the account currently executing it, verify that only the account that is delegating execution is the one that has its balance modified.
+        - [ ] If the opcode modifies the code of the account currently executing it, verify that only the account that is delegating execution is the one that has its code modified.
+    - [ ] Code call
+    - [ ] Initcode
+        - [ ] Verify opcode behaves as expected when running during the initcode phase of contract creation
+            - [ ] Initcode of a contract creating transaction.
+            - [ ] Initcode of a contract creating opcode (including itself if opcode creates a contract).
```

Comment:
So CREATE and CREATE2.

Also be aware of weird quirks like contract A doing CREATE2, then in this create context calling back into A, which then attempts to deploy exactly the same CREATE2 (so tries to deploy at same address. This is prevented because the nonce of the to-be-created address gets bumped to 1 prior to executing, and prior to trying to create a new address the nonce of the target address should be 0, so the second attempt will immediately fail).

Reply:
Added:
> Verify recursive contract creation using the opcode: Factory contract uses the opcode, and initcode calls back to factory contract

To the `Contract creation section below.